### PR TITLE
Fix for #407: Show number of hardlinks for lsd -l

### PR DESCRIPTION
**Pull request by TheAlakazam**
+ Added implementation for displaying number of hardlinks to a file.
+ Displays '-' in windows
---
#### TODO

- [x] Use `cargo fmt`
- [x] Add necessary tests
- [x] Add changelog entry

### CODE REVIEW

File: src/color.rs [hunk 1]
```diff
@@ -243,6 +251,8 @@ impl Colors {
         // INode
         m.insert(Elem::INode { valid: true }, Colour::Fixed(13)); // Pink
         m.insert(Elem::INode { valid: false }, Colour::Fixed(245)); // Grey
+        m.insert(Elem::INodeCount { valid: true }, Colour::Fixed(134));
```
Comment (meain) [inline src/color.rs:254]: It is better to use colors <16. I think we can reuse `13` and `245` here.

File: src/flags/blocks.rs [hunk 2]
```diff
@@ -136,7 +136,8 @@ impl Blocks {
 
     /// Prepends a [Block] of variant [INode](Block::INode) to `self`.
     fn prepend_inode(&mut self) {
-        self.0.insert(0, Block::INode)
+        self.0.insert(0, Block::INode);
+        self.0.insert(1, Block::INodeCount)
```
Comment (meain) [inline src/flags/blocks.rs:140]: As of now, I think we can have `inode-count` enabled only when they specify it explicitly in blocks. Let `--inode` only include the `inode` block.

File: src/flags/blocks.rs [hunk 3]
```diff
@@ -181,6 +183,7 @@ impl TryFrom<&str> for Block {
             "date" => Ok(Self::Date),
             "name" => Ok(Self::Name),
             "inode" => Ok(Self::INode),
+            "inode_count" => Ok(Self::INodeCount),
```
Comment (meain) [inline src/flags/blocks.rs:186]: `inode-count`.

**Review by meain — COMMENTED** [review]
Looks good, a few minor changes.

**Comment by meain** [comment]
I was thinking we could change the name to `links`. Single word name would be better in this case.

File: src/flags/blocks.rs (hunk 2 above)
Comment (TheAlakazam) [reply src/flags/blocks.rs:140]: But the number of hard links is only show when the inode option is selected they, are tied in together. I could seperate and create a `prepend_inode_count()` method to do so and call it after the `prepend_inode()` method, would that be alright?

Comment (meain) [reply src/flags/blocks.rs:140]: What I meant was we can leave the `--inode` flag as is and have `links`(`inode-count`) show up only when it is passed explicitly in `--blocks`. Something like `lsd --blocks name,size,links`.

Comment (TheAlakazam) [reply src/flags/blocks.rs:140]: I had a doubt, `links` should only be shown with the long option or also as is when passed to blocks.

Comment (meain) [reply src/flags/blocks.rs]: Actually both, but that is part of a separate logic and you don't really have to handle that. All you have to do is to add a new block and the logic to parse the `links` block in from config and cli. Something like [this](https://github.com/meain/lsd/blob/e3649fede8f7fc64041b709c56a3438ddbf813f0/src/flags/blocks.rs#L174)

File: src/flags/blocks.rs (hunk 3 above)
Comment (meain) [reply src/flags/blocks.rs]: When just long is passed, don't show it. I think we'll keep the current default long blocks as they are for now.

Comment (TheAlakazam) [reply src/flags/blocks.rs]: Right now it doesn't show if only long is passed, but only shows the links if blocks is passed.

**Comment by meain** [comment]
Similar to changing `inode_count` to `links`, can we also change the name `INodeCount` to `Links`.

File: src/color.rs (hunk 1 above)
Comment (meain) [reply src/color.rs:254]: Also, can we update the colors here to be similar to inode count. Adding colors over 16 might cause issues with the user's terminal colorscheme.

File: src/flags/blocks.rs [hunk 4]
```diff
@@ -124,6 +124,7 @@ impl Blocks {
             Block::Size,
             Block::Date,
             Block::Name,
+            Block::Links,
```
Comment (meain) [inline src/flags/blocks.rs:127]: No links by default
```suggestion
            Block::Name,
```

File: src/color.rs [hunk 5]
```diff
@@ -243,6 +251,8 @@ impl Colors {
         // INode
         m.insert(Elem::INode { valid: true }, Colour::Fixed(13)); // Pink
         m.insert(Elem::INode { valid: false }, Colour::Fixed(245)); // Grey
+        m.insert(Elem::Links { valid: true }, Colour::Fixed(134));
+        m.insert(Elem::Links { valid: false }, Colour::Fixed(222));
```
Comment (meain) [inline src/color.rs:255]: We could reuse the `Inode` colors
```suggestion
        m.insert(Elem::Links { valid: true }, Colour::Fixed(13));
        m.insert(Elem::Links { valid: false }, Colour::Fixed(245));
```

**Review by meain — CHANGES_REQUESTED** [review]
Just a few things that you missed.

**Comment by meain** [comment]
Awesome, thanks a lot!

**Comment by FischLu** [comment]
please add `links` option in the config.yaml example in the README.md file, today I tired to config my lsd with this function, but didn't see it anywhere till I see this pr

**Comment by meain** [comment]
Added in https://github.com/Peltoche/lsd/pull/808

**Pull request merged**